src/org.hdfgroup.hdfview/hdf/view/dialog/NewStringAttributeDialog.java

```diff
@@ -576,6 +593,23 @@ else if (tsize == 8) {
                     }
                     value = d;
                 }
+                else if (tsize == 16) {
+                    double[] d = new double[arraySize];
```

Comment:
Is this trying to use a `double[]` to implement support for `long double`? That's not going to work out correctly.